crates/router/src/types/api/payment_methods.rs

```diff
@@ -69,23 +69,23 @@ impl PaymentMethodCreateExt for PaymentMethodCreate {
                 self.payment_method_type,
             ),
             || {
-                return Err(report!(errors::ApiErrorResponse::InvalidRequestData {
+                Err(report!(errors::ApiErrorResponse::InvalidRequestData {
                     message: "Invalid 'payment_method_type' provided".to_string()
                 })
-                .attach_printable("Invalid payment method type"));
+                .attach_printable("Invalid payment method type"))
```

Comment:
Since you removed return, shouldn't this be propagated?

Reply:
Good catch, resolved.